app/src/combo.c

```diff
@@ -105,16 +107,37 @@ static int initialize_combo(struct combo_cfg *new_combo) {
 }
 
 static int setup_candidates_for_first_keypress(int32_t position, int64_t timestamp) {
+    // Due to layer filtering we need to track the candidates[index] stuff by hand
+    int number_of_combo_candidates = -1; // First increment moves this to 0 as expected
+    uint8_t highest_active_layer = zmk_keymap_highest_layer_active();
     for (int i = 0; i < CONFIG_ZMK_COMBO_MAX_COMBOS_PER_KEY; i++) {
+        number_of_combo_candidates++; // Increment on each pass of the loop
```

Comment:
I think we can clean this up a bit; 

initialize number_of_combo_candidates to 0, then remove this line and only increment it in the `if (combo_layer_valid)` clause.

Reply:
Re-worked the incrementing logic to simplify/cleanup